tests/phpunit/tests/user/getUserCount.php

```diff
@@ -0,0 +1,164 @@
+	}
+
+	/**
+	 * @ticket 22917
+	 * @group multisite
+	 * @group ms-required
+	 */
+	public function test_enable_live_network_user_counts_filter() {
+		// False for large networks by default.
+		add_filter( 'enable_live_network_counts', '__return_false' );
+
+		// Refresh the cache.
+		wp_update_network_counts();
+		$start_count = get_user_count();
+
+		wpmu_create_user( 'user', 'pass', 'email' );
+
+		// No change, cache not refreshed.
+		$count = get_user_count();
+
+		$this->assertSame( $start_count, $count );
+
+		wp_update_network_counts();
+		$start_count = get_user_count();
+
+		add_filter( 'enable_live_network_counts', '__return_true' );
+
+		self::factory()->user->create( array( 'role' => 'administrator' ) );
+
+		$count = get_user_count();
+		$this->assertEquals( $start_count + 1, $count );
+
+		remove_filter( 'enable_live_network_counts', '__return_false' );
+		remove_filter( 'enable_live_network_counts', '__return_true' );
+	}
+
+	/**
+	 *
+	 */
+	public function test_get_user_count_update() {
+		wp_update_user_counts();
+		$current_network_user_count = get_user_count();
+
+		self::factory()->user->create( array( 'role' => 'administrator' ) );
+
+		$user_count = get_user_count();
+
+		$this->assertEquals( $current_network_user_count + 1, $user_count );
+	}
+
+	/**
+	 * @group ms-excluded
+	 */
+	public function test_get_user_count_update_on_delete() {
+		wp_update_user_counts();
+		$current_network_user_count = get_user_count();
+
+		$u1 = self::factory()->user->create( array( 'role' => 'administrator' ) );
+
+		$user_count = get_user_count();
+
+		$this->assertEquals( $current_network_user_count + 1, $user_count );
+
+		wp_delete_user( $u1 );
+
+		$user_count_after_delete = get_user_count();
+
+		$this->assertEquals( $user_count - 1, $user_count_after_delete );
+	}
+
+	/**
+	 * @group ms-required
+	 */
+	public function test_get_user_count_update_on_delete_multisite() {
+		wp_update_user_counts();
+		$current_network_user_count = get_user_count();
+
+		$u1 = wpmu_create_user( 'user', 'pass', 'email' );
+
+		$user_count = get_user_count();
+
+		$this->assertEquals( $current_network_user_count + 1, $user_count );
+
+		wpmu_delete_user( $u1 );
+
+		$user_count_after_delete = get_user_count();
+
+		$this->assertEquals( $user_count - 1, $user_count_after_delete );
+	}
+
+	/**
+	 * @group multisite
+	 * @group ms-required
+	 */
+	public function test_get_user_count() {
+		// Refresh the cache.
+		wp_update_network_counts();
+		$start_count = get_user_count();
+
+		// Only false for large networks as of 3.7.
+		add_filter( 'enable_live_network_counts', '__return_false' );
+		self::factory()->user->create( array( 'role' => 'administrator' ) );
+
+		$count = get_user_count(); // No change, cache not refreshed.
+		$this->assertSame( $start_count, $count );
+
+		wp_update_network_counts(); // Magic happens here.
+
+		$count = get_user_count();
+		$this->assertEquals( $start_count + 1, $count );
```

Comment:
```suggestion
		$this->assertSame( $start_count + 1, $count );
```
`get_user_count()` returns `int`, so this assertion can be made stricter.